File: practice23/12.cpp

//https://www.nowcoder.com/practice/56f412b9e17e4504b1bbeca96f94a263?tpId=98&tqId=32876&tPage=3&rp=3&ru=/ta/2019test&qru=/ta/2019test/question-ranking
#include<iostream>
#include<string>
using namespace std;

int main()
{
	long long  x, y;
	while (cin >> x >> y)
	{
		long long m = x + y;
		long long n = y;
		long long sum1 = m, sum2 = n;
		if (y == 1)
		{
			cout << m << endl;
		}
		else
		{
			m--, n--;
			for (; y > 1; y--)
			{
				sum1 = sum1*m;
				sum2 = sum2*n;
				m--, n--;
			}
			cout << sum1 / sum2 << endl;
		}
	}
	system("pause");
	return 0;
}

File: practice32/12.java


//https://www.nowcoder.com/practice/876e3c5fcfa5469f8376370d5de87c06?tpId=85&tqId=29847&tPage=1&rp=1&ru=/ta/2017test&qru=/ta/2017test/question-ranking
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n=sc.nextInt();
        int[] pt=new int[n];
        for(int i=0;i<n;i++){
            pt[i]=sc.nextInt();
        }
        Arrays.sort(pt);
        int result=0;
        for(int i=0;i<n;i++){
            if(pt[i]>result+1){
                break;
            }
            result=result+pt[i];
        }
        System.out.println(result+1);
    }
}